### PR TITLE
feature: support video.js 6 forward compatibility

### PR DESCRIPTION
**Pull request by forbesjo**

### CODE REVIEW

**Comment by gesinger** [comment]
Though we should also just start using currentSource instead of using the sourceGrabber.

File: src/plugin.js [hunk 1]
```diff
@@ -82,7 +82,7 @@ let sourceGrabber = {
 eme.options = {};
 
 // register to beginning of HTML5 source handlers
-videojs.getComponent('Html5').registerSourceHandler(sourceGrabber, 0);
+videojs.getTech('Html5').registerSourceHandler(sourceGrabber, 0);
 
 // Register the plugin with video.js.
 videojs.plugin('eme', eme);
```
Comment (gkatsev) [inline src/plugin.js]: we should update this to be:
```js
const registerPlugin = videojs.registerPlugin || videojs.plugin;

registerPlugin('eme', eme);
```

so it won't log a warning.

**Comment by BrandonOCasey** [comment]
we have also been updating `.travis.yml` to run tests on `video.js` 5 and 6 on travis. Here is the config: https://github.com/brightcove/videojs-contextmenu/blob/master/.travis.yml. This usually involves adding `travisChrome` to karma as well. like this config does: https://github.com/brightcove/videojs-contextmenu/blob/master/test/karma.conf.js

**Comment by forbesjo** [comment]
Looks like this will require a little more work since the plugin replaces itself

**Comment by forbesjo** [comment]
This would require a major version bump since `player.eme.options` will no longer be available until `player.eme(options)` is called.

**Comment by misteroneill** [comment]
Based on my reading of the code, this plugin does not replace itself. It replaces the shared `eme.options` value each time it's initialized. Meaning it reflects only the most recently passed options. In other words, multiple players on a single page using this plugin will all share the same `eme.options`.

It's ugly, but to maintain backward compatibility (that is, always having `eme.options` be an object) you could do something like this _after_ registering the plugin:

```js
videojs.getComponent('Player').prototype.eme.options = {};
```

**Comment by forbesjo** [comment]
@misteroneill what's your opinion of my latest commit?

**Comment by misteroneill** [comment]
My gut reaction is the source handler method was a bit safer. This assumes the source is set when the player is ready, which I'm hesitant to say is a guarantee. @gkatsev might have a different opinion.

**Comment by misteroneill** [comment]
We discussed in grooming and because this is using a `ready` event listener (rather than a `ready()` callback) it should work fine. Carry on! :+1:

**Comment by forbesjo** [comment]
@gesinger are you ok with https://github.com/videojs/videojs-contrib-eme/pull/12#issuecomment-278761771

**Comment by gesinger** [comment]
@forbesjo , yeah, I think we're fine getting rid of the empty objects piece now.

**Pull request merged**